### PR TITLE
Add prettier code formatting  and eslint

### PR DESCRIPTION
**Pull request by jason-fox**
Adds [prettier](https://prettier.io) and  [husky](https://github.com/typicode/husky). All JavaScript files are autoformated on commit using the Git Hooks hook. 

running `npm run prettier` will format all files with standardized whitespace.

Added [eslint](https://eslint.org/) as a JavaScript linter

running `npm run lint` will check the linter rules. This is also set-up on Travis

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	],
+    "extends": "standard",
+    "rules": {
+        "semi": [2, "always"],
+        "indent": 0,
+        "mocha/no-exclusive-tests": "error",
+        "space-before-function-paren": 0,
+
+        /* These should be removed */
+        "no-throw-literal" : 0,
```
Comment (jason-fox) [inline .eslintrc.json]: These rules here are additional exclusions to get the linter to run successfully even though they are valid complaints. Someone should remove this section and **fix** the relevant errors.

File: .travis.yml [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
 after_success: "cat ./coverage/lcov.info | ./node_modules/coveralls/bin/coveralls.js"
 branches:
   only:
     - develop
+
+install:
+ - npm install
+
+before_script:
+ - npm run lint
```
Comment (jason-fox) [inline .travis.yml]: This runs the linter

File: package.json [hunk 3]
```diff
@@ -32,21 +39,29 @@
...
     "jasmine": "2.9.0",
-    "nock": "3.1.0",
-    "proxyquire": "1.7.3",
-    "coveralls": "^3.0.0"
+    "lint-staged": "~7.3.0",
+    "nock": "^10.0.6",
+    "proxyquire": "1.7.3"
   },
   "scripts": {
+    "lint": "eslint ./lib --ext .js",
```
Comment (jason-fox) [inline package.json]: Ideally this should be `eslint ./lib ./test --ext .js`, but the tests have a lot of errors.
Once again this is something to be fixed at some point.

**Comment by fdelavega** [comment]
Thanks Jason!

**Pull request merged**